RTNeural/activation/activation_eigen.h

```diff
@@ -464,6 +464,418 @@ class PReLUActivationT
+    static constexpr auto out_size = size;
+
+    SoftplusActivationT()
+        : outs(outs_internal)
+    {
+        outs = v_type::Zero();
+    }
+
+    std::string getName() const noexcept { return "softplus"; }
+
+    constexpr bool isActivation() const noexcept { return true; }
+
+    RTNEURAL_REALTIME void reset() { }
+
+    /** Forward propagation for Softplus activation. */
+    RTNEURAL_REALTIME inline void forward(const v_type& ins) noexcept
+    {
+        /** Apply Softplus: ln(1 + exp(x)) */
+        outs = (1 + MathsProvider::exp(ins.array()).log(); //log() waiting for mathsprovider
+    }
+
+    Eigen::Map<v_type, RTNeuralEigenAlignment> outs;
+
+private:
+    T outs_internal alignas(RTNEURAL_DEFAULT_ALIGNMENT)[out_size];
+};
+
+/** Dynamic implementation of a Mish activation layer. */
+template <typename T, typename MathsProvider = DefaultMathsProvider>
+class MishActivation : public Activation<T>
+{
+public:
+    /** Constructor for Mish activation layer depending on the softplus activation internally */
+    explicit MishActivation(int size)
+        : Activation<T>(size, {}, "mish"), softplus(size)
+    {
+        inVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+        outVec = Eigen::Matrix<T, Eigen::Dynamic, Eigen::Dynamic>::Zero(size, 1);
+    }
+
+    MishActivation(std::initializer_list<int> sizes)
+        : MishActivation(*sizes.begin())
+    {
+    }
+
+    RTNEURAL_REALTIME inline void forward(const T* input, T* out) noexcept override
+    {
+        inVec = Eigen::Map<const Eigen::Matrix<T, Eigen::Dynamic, 1>, RTNeuralEigenAlignment>(
+            input, Layer<T>::in_size, 1);
+        /** mish = tanh(softplus(x)) */
+        Eigen::Matrix<T, Eigen::Dynamic, 1> softplusOut(Layer<T>::in_size);
+        softplus.forward(input, softplusOut.data());
+
+        outVec = MathsProvider::tanh(inVec.array() * softplusOut.array());
+
+        std::copy(outVec.data(), outVec.data() + Layer<T>::in_size, out);
+    }
+
+private:
+    Eigen::Matrix<T, Eigen::Dynamic, 1> inVec;
+    Eigen::Matrix<T, Eigen::Dynamic, 1> outVec;
+    SoftplusActivation<T, MathsProvider> softplus; // Softplus instance
+};
+
+/** Static implementation of a Mish activation layer. */
+template <typename T, int size, typename MathsProvider = DefaultMathsProvider>
+class MishActivationT
+{
+    using v_type = Eigen::Matrix<T, size, 1>; 
+
+public:
+    static constexpr auto in_size = size;
+    static constexpr auto out_size = size;
+
+    MishActivationT()
+        : outs(outs_internal), softplus()
+    {
+        outs = v_type::Zero();
+    }
+
+    std::string getName() const noexcept { return "mish"; }
+
+    constexpr bool isActivation() const noexcept { return true; }
+
+    RTNEURAL_REALTIME void reset() { }
+
+    /** Forward propagation for Mish activation. */
+    RTNEURAL_REALTIME inline void forward(const v_type& ins) noexcept
+    {
+        v_type softplusOut = softplus.forward(ins); // Apply Softplus
```

Comment:
I don't think `softplus.forward()` will return anything, so this probably won't compile? After calling `softplus.forward()`, you can access the results from `softplus.outs`.

Reply:
Sorry my mistake on this, this was scratch code, that i was meant to fix. I must have missed it after scrolling through the code too many times haha, I wonder if theres a better strategy for this.